packages/tools/client-debugger/client-debugger/src/Registry.ts

```diff
@@ -122,6 +125,41 @@ export class DebuggerRegistry extends TypedEventEmitter<DebuggerRegistryEvents>
 			this.postRegistryChange();
 			return true;
 		},
+		["CONNECT_CONTAINER"]: (untypedMessage) => {
+			const message = untypedMessage as ConnectContainerMessage;
+			const { containerId } = message.data;
+			const existingDebugger = this.registeredDebuggers.get(containerId);
+			if (existingDebugger === undefined) {
+				console.warn(`No active debugger found for container ID "${containerId}".`);
+				return false;
+			}
+			existingDebugger.container?.connect();
+			return true;
+		},
+		["DISCONNECT_CONTAINER"]: (untypedMessage) => {
+			const message: DisconnectContainerMessage =
+				untypedMessage as DisconnectContainerMessage;
+			const { containerId } = message.data;
+			const existingDebugger = this.registeredDebuggers.get(containerId);
+			if (existingDebugger === undefined) {
+				console.warn(`No active debugger found for container ID "${containerId}".`);
+				return false;
+			}
+			existingDebugger.container?.disconnect(/* TODO: Specify debugger reason here once it is supported */);
```

Comment:
I think we can start by just hardcoding a string here for now, so when the container disconnects, the telemetry makes it clear it was a deliberate action by the user. `Container was manually disconnected with the debugger.`. Same thing for closing.

Reply:
The APIs don't currently accept a reason string at all. We would need to update the IContainer connect/disconnect APIs to accept one. This is something I think we should definitely do, but outside of the scope of this PR.

Please correct me if I'm missing something :)

Reply:
Ah, you're correct. Definitely outside the scope of this PR. And while `close()` does support a reason, it's supposed to be passed only in case of error, which isn't the case here... Yeah, let's use the current comments for now.